src/mca/rmaps/base/rmaps_base_support_fns.c

```diff
@@ -664,9 +700,225 @@ prte_node_t *prte_rmaps_base_get_starting_point(pmix_list_t *node_list, prte_job
     }
 
 process:
-    PRTE_OUTPUT_VERBOSE((5, prte_rmaps_base_framework.framework_output, "%s Starting at node %s",
+    PRTE_OUTPUT_VERBOSE((5, prte_rmaps_base_framework.framework_output,
+                         "%s Starting at node %s",
                          PRTE_NAME_PRINT(PRTE_PROC_MY_NAME),
                          ((prte_node_t *) cur_node_item)->name));
 
-    return (prte_node_t *) cur_node_item;
+    /* put this node at the front of the list */
+    pmix_list_remove_item(node_list, cur_node_item);
+    pmix_list_prepend(node_list, cur_node_item);
```

Comment:
While debugging the PGI issue I saw the allocated_nodes list become corrupted after calling the `prte_rmaps_base_get_starting_point` function at the bottom of the `prte_rmaps_base_get_target_nodes` function . These two lines seemed to be the cause.

This fixed the PGI issue, but I'm a bit afraid that it is masking a different issue.
```suggestion
    if (pmix_list_get_first(node_list) != cur_node_item) {
        pmix_list_remove_item(node_list, cur_node_item);
        pmix_list_prepend(node_list, cur_node_item);
    }
```

Can you give me a scenario where you would need to execute this section of code to move the node to the front of the list?

Reply:
I dug thru the old PRRTE list functions and compared them to the PMIx functions we now use. There were only two differences:

1. PRRTE uses atomics to modify the refcount and PMIx does not. I don't think this is the issue as we only typically access lists from inside events. Still, something to keep in mind.
2. PMIx was missing a couple of type casts. These were right in the "remove_first" and "remove_last" inline functions, which means they _might_ have had an impact here. I added the type casts in https://github.com/openpmix/openpmix/pull/2642.

Let's see if that makes a difference - if not, then I'll add your fix.